### ISSUE TITLE
New console has a few shortcomings

### ISSUE THREAD

**Issue by jebej** [JunoLab/Juno.jl#102]
1. It seems history is not saved in the new console
2. "links" to julia file when using `@which` are no longer clickable, this was a pretty useful feature

**Comment by xximranxx** [comment]
A restart of the julia session does not seem to load .juliarc
To me it would make sense to reload .juliarc e.g. when changing git branches (and .juliarc containing `include` code)

**Comment by pfitzseb** [comment]
> It seems history is not saved in the new console

It is for me. Can you check `Base.REPL.find_hist_file()` in Juno and in a plain REPL?

> "links" to julia file when using `@which` are no longer clickable, this was a pretty useful feature

Yeah, that's unfortunate. You can usually click on links, but IIRC they need to start with `./` or similar (like in error messages). For that specific thing you can also just call `@edit sin(2)` or similar which will take you straight to the right file.

> A restart of the julia session does not seem to load .juliarc

It does for me ;)

> To me it would make sense to reload .juliarc e.g. when changing git branches (and .juliarc containing include code)

That seems like a bad idea to me. You can just manually kill Julia when doing that if you really need to.

**Comment by MikeInnes** [comment]
Some of these are probably due to our cycler booting a process before you ask for it (and therefore before the changes you made to history or .juliarc).

**Comment by jebej** [comment]
> Some of these are probably due to our cycler booting a process before you ask for it (and therefore before the changes you made to history or .juliarc).

Yes, this is what is happening. If I restart Atom, the "recent" history is there.

**Comment by pfitzseb** [comment]
If you don't mind waiting longer for new processes you could just switch to the Basic boot mode, which should alleviate those issues.

**Comment by jebej** [comment]
I've reverted back to the legacy console. There are other issues with the regular REPL I have noticed:
- you can't click to move the cursor, this was useful when entering multiple lines at a time
- you can't collapse or expand results
- types that don't have a dedicated print method still printed nicely in the old command, as it just showed a list of fields, which you could expand or collapse (in the REPL, all the field contents are printed inline, which is much less useful)
- issues with font printing with some Unicode characters, where the character width is not computed properly, making some characters overlap or the cursor be in the wrong place
- no more autocomplete suggestions
I have added those to the original issue.

What were the shortcomings of the original console? I feel like using a regular Julia REPL will be very limiting...

**Comment by jullit31** [comment]
Are these features going to be reimplemented? Other than 1. and 6., which are obviously bugs.
I guess we can't use the Legacy Console forever, can we?
